todoComments/todoComments.md

```diff
@@ -0,0 +1,185 @@
+| 82 | BANDAID FIX Application wasn't loading due to store.getState() returning undefined after adding call to translation in GraphicRateMenuComponent | src/client/app/utils/translate.ts | Needs Issue Created | - | Not related to any other translation-related issues. May need its own issue created. |
+| 83 | Its a bad practice to import store anywhere other than index.tsx migrate to useTranslate() from componentHooks.ts | src/client/app/utils/translate.ts | Needs Issue Created | - | See line 80 |
+| 84 | This should be moved to the error case below once issues with this process are under control. If all is well it could go inside the case where an unknown error occurred. | src/scripts/installOED.sh | Unsure | [Github issue #584](https://github.com/OpenEnergyDashboard/OED/issues/584) | Could possibly add this todo into the linked issue |
+| 85 | Verify that user see the message returned, see https://express-rate-limit.mintlify.app/reference/configuration#message | src/server/app.js | Issue Is Created | [Github issue #1206](https://github.com/OpenEnergyDashboard/OED/issues/1206) | Directly related to issue 1206 |
+| 86 | This was causing tests to fail for 3D rejection. This limit seems to be okay but we should find a better solution than upping values just for tests. | src/server/app.js | Needs Issue Created | - | Could not find any related issues. Underlying issues seems to be with the automatic test |
+| 87 | Potentially modify the Morgan logger to use the log API, thus unifying all our logging. | src/server/app.js | Needs Issue Created | - | No related issues |
+| 88 | Check if valid env variables are actually loaded despite the lack of a file, only log if they are not | src/server/config.js | Needs Issue Created | - | TODO is self-explanatory. No implementation in code yet. Needs an issue created. |
+| 89 | These likely need to be updated. | src/server/data/automatedTestingData.js | Unsure | - | References a series of commands to remove all the readings and meters associated withthe test data. This is valuable if you want to run the process again and the<br>meters and readings already exist. No related issues found. |
+| 90 | When this is converted to key/value pairs, the area should be set to the comparable meter(s) and campus is the sum of all 121,000. | src/server/data/websiteData.js | Unsure | - | - |
+| 91 | TODO | src/server/migrations/0.7.0-0.8.0/sql/readings/replace_compressed_readings_2.sql | Related To An Issue | - | TODO states that it is reliant on "having sites pick a site-level reading frequency" which is not done (yet?). I could not find an associated issue for that so not sure what to do with this for now. |
```

Comment:
Actually, this is obsolete as this functionality was added later. I had not realized until I saw this overview doc. If you do a PR to fix TODO items you are welcome to remove this.

Reply:
Just to double check, this TODO contains the following 3 lines of text. Is it okay to remove all three lines?

| TODO |
		- Note that the above documentation does not account for the increase in data when packaged in JSON format.
		- This function is complicated by the fact that groups can contain meters with different levels of granularity.
		We address this now by having sites pick a site-level reading frequency. (Not done)
		- When resource generalization is implemented in OED, this function will need to be done in a 
		meter-by-meter basis, since different resource meters will have different levels of granularity.

Reply:
Your comment prompted me to check the code more carefully. Here is my current thinking:

- Note that the above documentation does not account for the increase in data when packaged in JSON forma
  - This is still active so it stays. It relates to the sizes discussed vs what is actually transmitted.
- This function is complicated by the fact that groups can contain meters with different levels of granularity.
We address this now by having sites pick a site-level reading frequency. (Not done)
  - This is now obsolete and can be removed. There are two reasons. First, meter/raw data cannot be used for groups. Second, the SQL now adjusts if a meter within a group has a higher frequency of reading than desired.
- When resource generalization is implemented in OED, this function will need to be done in a
meter-by-meter basis, since different resource meters will have different levels of granularity.
  - This is obsolete/done so can be removed.